### PR TITLE
chore: adaptations for nightly-2024-02-01

### PR DESCRIPTION
**Pull request by kim-em**
This is the adaptation PR for nightly-2024-02-01.

In rolls in the branches
* #9843, prepared by @mattrobball, which has the adaptations for leanprover/lean4#2478
* #10133, prepared by @semorrison, which has the adaptations for leanprover/lean4#3210

As these both landed in the same nightly, we're having to do the update in one go.

Note this nightly is intended to become `v4.6.0-rc1` tomorrow.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/WithZero.lean [hunk 1]
```diff
@@ -281,8 +281,10 @@ theorem OrderIso.mulRight₀'_symm {a : α} (ha : a ≠ 0) :
   rfl
 #align order_iso.mul_right₀'_symm OrderIso.mulRight₀'_symm
 
+set_option synthInstance.maxHeartbeats 0 in
+set_option maxHeartbeats 0 in
```
Comment (kbuzzard) [inline Mathlib/Algebra/Order/WithZero.lean:285]: I was surprised to see these here. Do we want more concrete bounds?

**Review by kbuzzard — APPROVED** [review]
The diff looks pretty harmless to me (other than the tactic stuff, which I don't understand) but there are a couple of funny `set_option`s and I don't know if this is intentional (edit: it almost certainly isn't; I've just removed them and things work fine)

**Comment by mattrobball** [comment]
Thanks. I looked over the changes last night from #9843 and everything looked good except that. I thought I push a fix though.

**Review by mattrobball — COMMENTED** [review]
Some unnecessary lines.

File: Mathlib/RingTheory/DedekindDomain/Ideal.lean [hunk 2]
```diff
@@ -598,9 +598,9 @@ a computable alternative.
 -- Porting note: added noncomputable because otherwise it fails, so it seems that the goal
 -- is not achieved...
 noncomputable instance FractionalIdeal.cancelCommMonoidWithZero :
-    CancelCommMonoidWithZero (FractionalIdeal A⁰ K) :=
-  { @FractionalIdeal.commSemiring A _ A⁰ K _ _,  -- Project out the computable fields first.
-    (by infer_instance : CancelCommMonoidWithZero (FractionalIdeal A⁰ K)) with }
+    CancelCommMonoidWithZero (FractionalIdeal A⁰ K) := inferInstance
+  -- { @FractionalIdeal.commSemiring A _ A⁰ K _ _,  -- Project out the computable fields first.
```
[the remark this answers is not in the document]
Comment (mattrobball) [reply Mathlib/RingTheory/DedekindDomain/Ideal.lean]: Not removed, but merged with the porting note.

File: Mathlib/Algebra/Lie/Subalgebra.lean [hunk 3]
```diff
@@ -193,7 +193,6 @@ theorem mk_coe (S : Set L) (h₁ h₂ h₃ h₄) :
   rfl
 #align lie_subalgebra.mk_coe LieSubalgebra.mk_coe
 
-@[simp]
```
Comment (mattrobball) [inline Mathlib/Algebra/Lie/Subalgebra.lean:196]: Removed because `simp` can solve this

Comment (mattrobball) [inline Mathlib/Algebra/Lie/Subalgebra.lean:196]: Removed because `simp` can solve this

File: Mathlib/Algebra/Lie/Submodule.lean [hunk 4]
```diff
@@ -128,7 +128,6 @@ theorem coe_toSet_mk (S : Set M) (h₁ h₂ h₃ h₄) :
   rfl
 #align lie_submodule.coe_to_set_mk LieSubmodule.coe_toSet_mk
 
-@[simp]
```
Comment (mattrobball) [inline Mathlib/Algebra/Lie/Submodule.lean:131]: Removed because `simp` can solve this

**Review by mattrobball — APPROVED** [review]

File: Mathlib/Tactic/NormNum/Core.lean [hunk 5]
```diff
@@ -189,10 +189,12 @@ initialize registerBuiltinAttribute {
     modifyEnv fun env => normNumExt.modifyState env fun _ => s
 }
 
-/-- A simp plugin which calls `NormNum.eval`. -/
-def tryNormNum? (post := false) (e : Expr) : SimpM (Option Simp.Step) := do
-  try return some (.done (← eval e post))
-  catch _ => return none
+/-- A simplifier step for `norm_num`. -/
```
Comment (semorrison) [inline Mathlib/Tactic/NormNum/Core.lean:192]:
```suggestion
/-- A simp plugin which calls `NormNum.eval`. -/
```

File: Mathlib/RingTheory/DedekindDomain/Ideal.lean [hunk 6]
```diff
@@ -596,11 +596,11 @@ Although this instance is a direct consequence of the instance
...
 -- Porting note: added noncomputable because otherwise it fails, so it seems that the goal
--- is not achieved...
+-- is not achieved... Previous attempt was
+-- { @FractionalIdeal.commSemiring A _ A⁰ K _ _,  -- Project out the computable fields first.
+--   (by infer_instance : CancelCommMonoidWithZero (FractionalIdeal A⁰ K)) with }
 noncomputable instance FractionalIdeal.cancelCommMonoidWithZero :
-    CancelCommMonoidWithZero (FractionalIdeal A⁰ K) :=
-  { @FractionalIdeal.commSemiring A _ A⁰ K _ _,  -- Project out the computable fields first.
-    (by infer_instance : CancelCommMonoidWithZero (FractionalIdeal A⁰ K)) with }
+    CancelCommMonoidWithZero (FractionalIdeal A⁰ K) := inferInstance
```
Comment (eric-wieser) [inline Mathlib/RingTheory/DedekindDomain/Ideal.lean:603]: #10173 contains a fix for this that works on master and this branch.

File: Mathlib/RepresentationTheory/GroupCohomology/Hilbert90.lean [hunk 7]
```diff
@@ -102,6 +102,6 @@ noncomputable instance hilbert90 : Unique (H1 (Rep.ofAlgebraAutOnUnits K L)) whe
     refine' Additive.toMul.bijective.1 _
     show Units.map g β⁻¹ / β⁻¹ = Additive.toMul (x.1 g)
     rw [map_inv, div_inv_eq_mul, mul_comm]
-    exact mul_inv_eq_iff_eq_mul.2 (hβ g).symm
+    apply mul_inv_eq_iff_eq_mul.2 (hβ g).symm
```
Comment (eric-wieser) [inline Mathlib/RepresentationTheory/GroupCohomology/Hilbert90.lean:105]: Why does `exact` fail here?

Comment (mattrobball) [reply Mathlib/RepresentationTheory/GroupCohomology/Hilbert90.lean:105]: My guess is transparency

Comment (semorrison) [inline Mathlib/RepresentationTheory/GroupCohomology/Hilbert90.lean:105]:
```suggestion
    -- TODO this used to be `exact` prior to leanprover/lean4#2478
    apply mul_inv_eq_iff_eq_mul.2 (hβ g).symm
```

**Pull request merged**